test/e2e/framework/node/helper.go

```diff
@@ -204,10 +205,15 @@ func RemoveExtendedResource(ctx context.Context, clientSet clientset.Interface,
 	extendedResource := v1.ResourceName(extendedResourceName)
 
 	ginkgo.By("Removing a custom resource")
-	node, err := clientSet.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{})
-	framework.ExpectNoError(err)
-	delete(node.Status.Capacity, extendedResource)
-	delete(node.Status.Allocatable, extendedResource)
-	_, err = clientSet.CoreV1().Nodes().UpdateStatus(ctx, node, metav1.UpdateOptions{})
+	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
No need to retry I think. You can just do patch with status subresource.

Reply:
But, you intentionally used the update, not patch, for ease here, right?
https://github.com/kubernetes/kubernetes/pull/128194#discussion_r1829913783

Reply:
Yeah. I thought there won't be resource conflict and hence used UpdateStatus. We can revert to patch again to avoid doing retry.

Reply:
I meant, that was for the implementation simplicity, right?
I'm not sure why you prefer to switch back to the patch here. Efficiency-wise too, the flake is pretty rare, that is, the retry would happen only rarely, and I don't think we need to mind (... and this is a test code either way).

Reply:
this is an e2e test so we don't need to over optimize, I think current approach is ok and avoids more code churn

Reply:
> I meant, that was for the implementation simplicity, right? I'm not sure why you prefer to switch back to the patch here. Efficiency-wise too, the flake is pretty rare, that is, the retry would happen only rarely, and I don't think we need to mind (... and this is a test code either way).

patch requests avoid resource conflicts. so there won't be any need to retry and we can keep the logic similar to `AddExtendedResource` func above.

Reply:
either way works for me, I also prefer to use patch if the patch is really simple ... I'm just trying to say we should not overthink it, chose one solution and move on , a retry on conflict is common in multiple places of the code, so is nota bad solution either

Reply:
We need either retry on update, or simply patch. Reported by @wendy-ha18 we got flakes and it's impacting 1.32 release: https://kubernetes.slack.com/archives/C09TP78DV/p1733547680436909

Agree with @aojea we should move on either way. (we can follow-up with the approach of switching to patch)